streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/Graph.java

```diff
@@ -117,6 +131,8 @@ private Graph(final boolean isResidualGraph) {
     }
 
     public void addEdge(final V u, final V v, final int capacity, final int cost, final int flow) {
+        Objects.requireNonNull(u);
+        Objects.requireNonNull(v);
```

Comment:
Why do we need these checks?

Reply:
`null` is a special node we introduced in graph as dummy source to detect negative cycle. So users can use it